homeassistant/components/climate/warmup4ie.py

```diff
@@ -0,0 +1,202 @@
+        """Return the current temperature."""
+        return self._device.get_current_temmperature()
+
+    @property
+    def target_temperature(self):
+        """Return the temperature we try to reach."""
+        return self._device.get_target_temmperature()
+
+    @property
+    def min_temp(self):
+        """Return the minimum temperature."""
+        return self._device.get_target_temperature_low()
+
+    @property
+    def max_temp(self):
+        """Return the maximum temperature."""
+        return self._device.get_target_temperature_high()
+
+    @property
+    def is_away_mode_on(self):
+        """Return if away mode is on."""
+        return self._away
+
+    @property
+    def is_on(self):
+        """Return true if the device is on."""
+        return self._on
+
+    @property
+    def operation_list(self):
+        """Return the operation modes list."""
+        return self._operation_list
+
+    @property
+    def current_operation(self):
+        """Return current operation ie. manual, auto, frost."""
+        return self._current_operation_mode
+
+    def set_temperature(self, **kwargs):
+        """Set new target temperatures."""
+        if kwargs.get(ATTR_TEMPERATURE) is not None:
+            self._device.set_new_temperature(kwargs.get(ATTR_TEMPERATURE))
+
+    def turn_away_mode_on(self):
+        """Turn away mode on."""
+        self._away = True
+        self._device.set_location_to_frost()
+
+    def turn_away_mode_off(self):
+        """Turn away mode off."""
+        self._away = False
+        self._device.set_temperature_to_manual()
+
+    def set_operation_mode(self, operation_mode):
+        """
+        Set new target operation mode.
+
+        Switch device on if was previously off
+        """
+        if not self.is_on:
+            self._on = True
+        if operation_mode == STATE_AUTO:
+            self._device.set_temperature_to_auto()
+            self._current_operation_mode = operation_mode
+            return
+        if operation_mode == STATE_MANUAL:
+            self._device.set_temperature_to_manual()
+            self._current_operation_mode = operation_mode
+            return
+
+    def turn_on(self):
+        """Turn on."""
+        self._on = True
+        self._device.set_temperature_to_manual()
+
+    def turn_off(self):
+        """Turn off."""
+        self._on = False
+        self._device.set_location_to_off()
+
+    def update(self):
+        """Fetch new state data for this device.
+
+        This is the only method that should fetch new data for Home Assistant.
+        """
+        if not self._device.update_room():
+            _LOGGER.error("updating Warmup4IE component failed.")
+
+        # set operation mode
+        mode_map = {'prog': STATE_AUTO, 'fixed': STATE_MANUAL}
+        self._current_operation_mode = mode_map.get(
+            self._device.get_run_mode(), STATE_MANUAL)
+
+        # set whether device is on/off
+        if self._device.get_run_mode() == 'off':
```

Comment:
Does `self._device.get_run_mode` return a cached result?

Reply:
Yes, this is a cached result. Fetching data from the thermostat is only done via self._device.update_room().